theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,118 @@
+# Custom Server Side Ad Insertion
+
+THEOplayer provides a way to integrate with third-party advertisement providers,
+and have them report their ads and ad-related events through the THEOplayer APIs.
+
+## Android SDK
+
+### Prerequisites
+
+Your THEOplayer SDK needs to be on version 7.6.0 or higher.
+
+### Integration ID
+
+To make use of the custom server side ad insertion, first you need an integration ID which will identify the provided sources and ads to the specific ad provider.
+
+### CustomSsaiDescriptionSerializer
```

Comment:
Move this *after* the `ServerSideAdIntegrationFactory`. This is more of a "chore" that you have to do if you're using a `CustomSsaiDescription`.

It also helps to link to an example, e.g. from [our Yospace connector](https://github.com/THEOplayer/android-connector/blob/8.8.3/connectors/yospace/src/main/java/com/theoplayer/android/connector/yospace/YospaceSsaiDescription.kt#L84-L125).

Reply:
True, but from implementation flow point, you need to have the `CustomSsaiDescriptionSerializer` setup to receive the source in the handler.

Reply:
It still feels backwards to me. Maybe first show how to subclass `ServerSideAdIntegrationFactory` and `CustomSsaiDescription`, and then afterwards go through the "chore" of implementing a `CustomSsaiDescriptionSerializer`?